gadget/gadget.go

```diff
@@ -250,6 +250,12 @@ type DiskVolumeDeviceTraits struct {
 	// the volume that may be useful in identifying whether a disk matches a
 	// volume or not.
 	Structure []DiskStructureDeviceTraits `json:"structure"`
+
+	// StructureEncryption is the set of partitions that are encrypted on the
+	// volume - this should only ever be ubuntu-data or ubuntu-save for now, but
+	// the map will indicate the name of the structure as the key and the type
+	// of encryption (currently only "LUKS") as the value in the map.
+	StructureEncryption map[string]map[string]string `json:"structure-encryption"`
```

Comment:
any reason not to use an actual struct for the entries?

```
type StructureEncryption struct {
    Method string `json:"method"`
}
```

or similar?

Reply:
No reason, just was quicker to do a map of strings like this, will change this then it also resolves the question about string constants and makes some of the validation logic simpler

Reply:
Yes actually let's take this in a followup as mentioned in the SU